apps/admin_api/lib/admin_api/v1/controllers/wallet_controller.ex

```diff
@@ -123,6 +122,26 @@ defmodule AdminAPI.V1.WalletController do
     do:
       handle_error(conn, :invalid_parameter, "Invalid parameter provided. `address` is required.")
 
+  defp add_balances_if_allowed({:error, code, description}, _), do: {:error, code, description}
+
+  defp add_balances_if_allowed(%Paginator{} = paginator, conn) do
+    with {:ok, %{query: query}} <- authorize(:view_balance, conn.assigns, nil),
+         true <- !is_nil(query) do
+      BalanceLoader.add_balances_to_wallets_intersect(paginator, query)
+    else
+      _ -> paginator
+    end
+  end
+
+  defp add_balances_if_allowed(wallet, conn) do
+    with {:ok, _} <- authorize(:view_balance, conn.assigns, wallet) do
+      {:ok, wallet} = BalanceLoader.add_balances(wallet)
```

Comment:
Should this go inside the `with` clause? Noticed `BalanceFetcher` will always return `{:ok, _}` but future-proofing...